lib/pbench/server/api/resources/endpoint_configure.py

```diff
@@ -0,0 +1,102 @@
+        """
+        self.logger = logger
+        self.host = config.get("pbench-server", "proxy_host")
+        uri_prefix = config.rest_uri
+        self.uri = f"{self.host}{uri_prefix}"
+        self.prefix = get_index_prefix(config)
+        self.commit_id = config.COMMIT_ID
+
+    def get(self):
+        """
+        Return server configuration information required by the UI
+        dashboard. This includes:
+
+        metadata: Information about the server configuration
+            result_index: The "root" index name for Pbench result data,
+                qualified by the current index version and prefix. In the
+                current ES schema, this is "v5.result-data-sample."
+            result_data_index: The "result-data" index has been broken into
+                "result-data-sample" and "result-data" indices for the
+                Elasticsearch V7 transition. In the current ES schema, this
+                is "v5.result-data."
+            run_index: The "master" run-data index root. In the current ES
+                schema, this is "v6.run-data."
+            run_toc_index: The Elasticsearch V7 index for run TOC data. In
+                the current ES schema, this is "v6.run-toc."
+        api:    A list of the server APIs supported; we give a name, which
+                corresponds to either the current config.json property name
+                (e.g., "elasticsearch") or the current src/service method name
+                being replaced (e.g., "queryControllers")
+            results: Direct access to Pbench data sets for the dashboard; this
+                is currently direct HTTP access through Apache public_html, and
+                not an "API" as such; we will need to adopt this as a real
+                server API when Pbench moves from local filesystem to S3 storage
+                schemas
+            elasticsearch: The Pbench pass-through URI for the Elasticsearch
+                cluster. This will eventually be superceded by the native
+                Pbench APIs, though it might remain accessible with special
+                user/group privileges to support special cases
+            graphql: The GraphQL frontend on postgreSQL currently used by the
+                dashboard user mocks. This will be superceded and deprecated
+                by native Pbench user management APIs
+            queryControllers: Return information about the run documents that
+                were created within a specified range of months.
+            queryMonthIndices: Return the YYYY-mm date strings for all ES
+                vx.run-data.* indices, in descending order
+
+        TODO: We need an internal mechanism to track the active versions of the
+        various Elasticsearch template documents. We're hardcoding them here and
+        in other APIs. We should consider persisting an equivalent of the
+        mapping table built in "indexer.py" for use across the server APIs.
```

Comment:
We want to move away from the Dashboard having to "know" how to construct Elasticsearch queries, so would this be really necessary once we add APIs for all the explicit queries made?

Reply:
Yes, after we've replaced all the APIs there should be no need for the dashboard to know anything about index names, which will be nice.